tools/driver/p4c_src/main.py

```diff
@@ -224,39 +307,152 @@ def main():
         if (pattern.match(opts.target + '-' + opts.arch)):
             backend = target
             break
-    if backend == None:
+    if backend is None:
         parser.error("Unknown backend: {}-{}".format(str(opts.target),
                                                      str(opts.arch)))
+    error_count = 0
+
+    JSON_input_specified = env_indicates_developer_build and opts.json_source
+### ^^^^^^^^^^^^^^^^^^^^: DRY
+    P4_input_or_inputs_specified = len(opts.P4_source_files) > 0
+    any_input_specified = JSON_input_specified or P4_input_or_inputs_specified
+### ^^^^^^^^^^^^^^^^^^^: for readability
 
-    # When using --help-* options, we don't necessarily need to pass an input file
-    # However, by default the driver checks the input and fails if it does not exist.
-    # In that case we set source to dummy.p4 so sanity checking works. Backend can
-    # force this behaviour for its own help options by overriding the
-    # should_not_check_input method.
+    if JSON_input_specified and P4_input_or_inputs_specified:
+        ### Note: this restriction was added by Abe, so if there is a test case
+        ###       which tries to feed the driver both a JSON input pathname
+        ###       via the relevant flag _and_ another input pathname,
+        ###       that will now fail whereas before the JSON input pathname
+        ###       would silently override the other input pathname[s].
+        print("\n"
+              "ERROR: it seems that both (firstly) a JSON input pathname "
+              'was specified via "' + JSON_input_flag + '"'
+              " _and_ (secondly) " + str(len(opts.P4_source_files)) + ""
+              ###                                                     ^^
+              ### The above nonsense is to get rid of a W504 in “pycodestyle”
```

Comment:
If pycodestyle is stupid you should probably just disable that rule. I assume it's possible via comments; for pylint it is.

Reply:
I made some changes to the defaults for “pycodestyle”; please see the file “setup.cfg___for_pycodestyle” in the same subdirectory.  [“setup.cfg” is merely a symlink to that file; I did it that way on purpose because IMO the name “setup.cfg” is too vague, and I didn`t want anybody thinking that I had checked garbage into the repo.]

Do you think I should add W504 to the “setup.cfg___for_pycodestyle” file and remove the relevant idiom from the Python code?